src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -328,17 +342,25 @@ def create_ave_image(ref_image: np.ndarray,
         ``"smooth_sigma_time"``
             Time Gaussian smoothing of frames to apply before correlation.
             Used as minimum start value in parameter search (float).
+    trim_frames_start : int, optional
+        Number of frames to disregard from the start of the movie. Default 0.
+    trim_frames_start : int, optional
+        Number of frames to disregard from the end of the movie. Default 0.
     batch_size : int, optional
-
+        Number of frames to process at once.
     """
     with h5py.File(suite2p_args['h5py']) as raw_file:
         frames_dataset = raw_file[suite2p_args['h5py_key']]
-        tot_frames = frames_dataset.shape[0]
+        tot_frames = (frames_dataset.shape[0]
+                      - trim_frames_end
+                      - trim_frames_start)
         ave_frame = np.zeros((frames_dataset.shape[1],
                               frames_dataset.shape[2]))
         dy_max = 0
         dx_max = 0
-        for start_idx in np.arange(0, tot_frames, batch_size):
+        for start_idx in np.arange(trim_frames_start,
+                                   tot_frames - trim_frames_end,
```

Comment:
Didn't you already handle this `tot_frames-trim_frames_end` above at line ~ 354 when you adjusted the value of `tot_frames`?

Reply:
Thanks for the catch. Needed to change `tot_frames` to make sure the averages were correct, missed the effect this would have on the indexing. Fixed it by using the dataset shape for iteration.